autoware_utils_debug/include/autoware_utils_debug/debug_publisher.hpp

```diff
@@ -0,0 +1,77 @@
+// Copyright 2020 Tier IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef AUTOWARE_UTILS_DEBUG__DEBUG_PUBLISHER_HPP_
+#define AUTOWARE_UTILS_DEBUG__DEBUG_PUBLISHER_HPP_
+
+#include "autoware_utils_debug/debug_traits.hpp"
+
+#include <rclcpp/publisher_base.hpp>
+#include <rclcpp/rclcpp.hpp>
+#include <rosidl_runtime_cpp/traits.hpp>
+
+#include <memory>
+#include <string>
+#include <unordered_map>
+
+namespace autoware_utils_debug
+{
+namespace debug_publisher
+{
```

Comment:
Other packages do not have nested namespaces, so wouldn’t it be a good opportunity to remove them? If there are compatibility issues, it would also be fine to work around them using `using`, just like the others.

Reply:
I think these are namespaces (debug_publisher and debug_traits) to hide implementations like `detail` or `impl`.
So how about moving it to `debug_traits` or making it a static private function of DebugPublisher class?

Reply:
Oh, I'm sorry about my misunderstand. Then there is no problem.

Reply:
The naming leads misunderstand though.